sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/models/DocumentResult.java

```diff
@@ -64,4 +69,19 @@ public TextAnalyticsError getError() {
     public boolean isError() {
         return isError;
     }
+
+    /**
+     * Throw a {@link TextAnalyticsException} if result has isError true and when a non-error property was accessed.
+     *
+     * @param logger the {@link ClientLogger} used to log the exception.
+     */
+    void throwExceptionIfError(ClientLogger logger) {
```

Comment:
Don't use a logger created in another class. For debugging purposes, we should know which class actually has the code that logged something.